expr/functions/moving/function.go

```diff
@@ -110,15 +111,42 @@ func (f *moving) Do(ctx context.Context, e parser.Expr, from, until int64, value
 	if err != nil {
 		return nil, err
 	}
+	if len(arg) == 0 {
+		return []*types.MetricData{}, nil
+	}
+
+	if len(e.Args()) >= 3 && e.Target() == "movingWindow" {
+		cons, err = e.GetStringArgDefault(2, "average")
+		if err != nil {
+			return nil, err
+		}
 
-	if len(e.Args()) == 3 {
+		if len(e.Args()) == 4 {
+			xFilesFactor, err = e.GetFloatArgDefault(2, float64(arg[0].XFilesFactor))
```

Comment:
I think the index should be 3 in this case:

```suggestion
			xFilesFactor, err = e.GetFloatArgDefault(3, float64(arg[0].XFilesFactor))
```